app/src/main/java/com/ataulm/artcollector/Navigation.kt

```diff
@@ -9,7 +9,10 @@ private const val GALLERY = "${BuildConfig.APPLICATION_ID}.gallery.ui.GalleryAct
 private const val ARTIST_GALLERY = "${BuildConfig.APPLICATION_ID}.artist.ui.ArtistActivity"
 private const val PAINTING = "${BuildConfig.APPLICATION_ID}.painting.ui.PaintingActivity"
 
-fun webIntent(webUrl: String) = Intent(Intent.ACTION_VIEW).setData(Uri.parse(webUrl))
+private const val INTENT_EXTRA_IMAGE_URL = "${BuildConfig.APPLICATION_ID}.IMAGE_URL"
+fun Intent.imageUrl(): String? = getStringExtra(INTENT_EXTRA_IMAGE_URL)
```

Comment:
I don't want to expose the keys across modules or even outside of this class.

I already felt it was a little bit cheating that GalleryActivity/DeepLinkActivity can know how to navigate to PaintingActivity (via this class) despite not having any dependency on the module PaintingActivity is in.

I thought it was due to a limitation in Android (with limited regex support in intent-filters, but now we're passing intent extras, there's even more coupling 😩 

I guess keeping them as simple as possible is best - the full URI for the PaintingActivity could still be expressed as a web link, with query parameters for the optional intent extras; yes... this way I can still blame the limited regex support 😅